src/connector/src/sink/utils.rs

```diff
@@ -68,12 +67,14 @@ pub(crate) mod dummy {
     #[allow(dead_code)]
     pub struct FeatureNotEnabledCoordinator<S: FeatureNotEnabledSinkMarker>(PhantomData<S>);
     #[async_trait::async_trait]
-    impl<S: FeatureNotEnabledSinkMarker> SinkCommitCoordinator for FeatureNotEnabledCoordinator<S> {
-        async fn init(&mut self, _subscriber: SinkCommittedEpochSubscriber) -> Result<Option<u64>> {
+    impl<S: FeatureNotEnabledSinkMarker> SinglePhaseCommitCoordinator
```

Comment:
Seems unnecessary to have this `FeatureNotEnabledCoordinator` after the associated type of the trait is removed.